### PR TITLE
6 post article comment- completed challenge 7 POST comment

### PR DESCRIPTION
**Pull request by Jookz**

### CODE REVIEW

File: __tests__/nc_news.test.js [hunk 1]
```diff
@@ -86,3 +86,42 @@ describe('GET /api/articles/:article_id', () => {
...
+            body: "Enjoyed it"
+        })
+        .then(({body}) => {
+            expect(body.comment).toHaveProperty("comment_id");
+            expect(body.comment).toHaveProperty("body");
+            expect(body.comment).toHaveProperty("article_id");
+            expect(body.comment).toHaveProperty("author");
+            expect(body.comment).toHaveProperty("votes");
+            expect(body.comment).toHaveProperty("created_at");
+
```
Comment (Jodiech) [inline __tests__/nc_news.test.js:113]: I'd consider adding an additional test that ignores unnecessary properties for example:
```
{
            username: "icellusedkars",
            body: "Enjoyed it",
            extra: "extra property"
        }
```
What status would this respond with?

File: __tests__/nc_news.test.js [hunk 2]
```diff
@@ -86,3 +86,42 @@ describe('GET /api/articles/:article_id', () => {
...
+        .send({
+            name: "icellusedkars",
+            body: "Enjoyed it"
+        })
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe("Bad request");
+        })
+    });
+});
```
Comment (Jodiech) [inline __tests__/nc_news.test.js:127]: A few more tests to add here please 🙏 make sure to test for:
- Invalid ID,  e.g. string of "not-an-id"
- Non existent ID, e.g. 0 or 9999
- If the username does not exist

Let me know if you need more info on this 👍

**Review by Jodiech — CHANGES_REQUESTED** [review]
Hey 👋 nice work on this ticket! I'd like you to add some more tests and resubmit when you're ready. Let me know if you need any help! 👍

File: __tests__/nc_news.test.js [hunk 3]
```diff
@@ -123,3 +123,218 @@ describe('GET /api/articles/:article_id', () => {
...
+        .then(({body}) => {
+            expect(body.comment).toHaveProperty("comment_id");
+            expect(body.comment).toHaveProperty("body");
+            expect(body.comment).toHaveProperty("article_id");
+            expect(body.comment).toHaveProperty("author");
+            expect(body.comment).toHaveProperty("votes");
+            expect(body.comment).toHaveProperty("created_at");
+
+        })
+    });
```
Comment (elijaharley) [inline __tests__/nc_news.test.js:183]: Good implementation here of the suggestion above. These tests cover all of the keys but not the values. You could make your tests more robust by adding a second argument to check the values, eg
` expect(body.comment).toHaveProperty("body", expect.any(String));`

File: __tests__/nc_news.test.js [hunk 4]
```diff
@@ -123,3 +123,218 @@ describe('GET /api/articles/:article_id', () => {
...
+            .send({
+                username: "BabeRuth",
+                body: "Enjoyed it"
+            })
+			.expect(404)
+			.then((response) => {
+				expect(response.body.msg).toBe('Entry not found');
+			});
+    });
+});
```
Comment (elijaharley) [inline __tests__/nc_news.test.js:220]: Good error testing here, I'd consider adding a test for an empty body also.

File: __tests__/nc_news.test.js [hunk 5]
```diff
@@ -123,3 +123,218 @@ describe('GET /api/articles/:article_id', () => {
...
+                  votes: 103,
+                  article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700'
+                }
+            const article = body.updated_article;
+            expect(article).toMatchObject(templateArticle);
+            
+
+        })
+    });
+});
```
Comment (elijaharley) [inline __tests__/nc_news.test.js:299]: This is good testing for PATCH. I like that you've considered that anything we'll need to test for POST we'll also likely need to test for PATCH.

File: __tests__/nc_news.test.js [hunk 6]
```diff
@@ -123,3 +123,218 @@ describe('GET /api/articles/:article_id', () => {
...
+                topic: 'mitch',
+                author: 'icellusedkars',
+                created_at: '2020-11-03T09:12:00.000Z',
+                article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700',
+                comment_count: 1,
+                votes: '0'
+              };
+            expect(body[0]).toMatchObject(articleAfterDeletion);
+        })
+    });
```
Comment (elijaharley) [inline __tests__/nc_news.test.js:323]: This test is a little confusing to read. 

We're making a DELETE request to '/api/comments/10' and then a request to GET all '/api/articles'. We don't necessarily need to use a GET request to demonstrate this here, but if you wanted to use a GET request to demonstrate that a comment is no longer linked to a specific article how could we demonstrate this more clearly?

**Review by elijaharley — APPROVED** [review]
Really good work on these tickets and updating to accommodate the feedback. I have left a couple of suggestions in your tests for you to consider before merging. Don't forget to .gitignore your playground files.

File: __tests__/nc_news.test.js (hunk 6 above)
Comment (Jookz) [reply __tests__/nc_news.test.js:323]: Used comment_count instead of whole object

**Review by RichardNeat — APPROVED** [review]
Nice one, good to go :)
